lib/encode.js

```diff
@@ -115,47 +101,45 @@ export async function encode({
     // generate legacy type table
     typeTable = createLegacyTypeTable({typeTable, appContextMap});
   } else {
+    // validate that an acceptable value for `registryEntryId` was passed
+    if(!(typeof registryEntryId === 'number' && registryEntryId >= 0)) {
+      throw new TypeError('"registryEntryId" must be a non-negative integer.');
+    }
+
     if(appContextMap !== undefined) {
       throw new TypeError(
-        '"appContextMap" must only be passed in "legacy" mode.');
+        '"appContextMap" must only be used with format "legacy-singleton".');
     }
     if(compressionMode !== undefined) {
       throw new TypeError(
-        '"compressionMode" must only be passed in "legacy" mode.');
-    }
-    if(typeTable !== undefined) {
-      if(typeTableLoader !== undefined) {
-        throw new TypeError('Use either "typeTable" or "typeTableLoader".');
-      }
-      if(!(typeTable instanceof Map)) {
-        throw new TypeError('"typeTable" must be a Map.');
-      }
-    } else if(registryEntryId === 1) {
-      // use default table (empty) for registry entry ID `1`
-      typeTable = createTypeTable({typeTable});
-    } else if(typeTableLoader) {
-      if(typeof typeTableLoader !== 'function') {
-        throw new TypeError('"typeTableLoader" must be a function.');
-      }
-      typeTable = await typeTableLoader({registryEntryId});
+        '"compressionMode" must only be used with format "legacy-singleton".');
     }
     if(registryEntryId !== 0) {
-      if(!typeTable) {
-        throw new CborldError(
-          'ERR_NO_TYPETABLE',
-          '"typeTable" not provided or found for "registryEntryId" ' +
-          `"${registryEntryId}".`);
-      }
-      // check to make sure unsupported types not present in `typeTable`
-      if(typeTable.has('http://www.w3.org/2001/XMLSchema#integer') ||
-        typeTable.has('http://www.w3.org/2001/XMLSchema#double') ||
-        typeTable.has('http://www.w3.org/2001/XMLSchema#boolean')) {
-        throw new CborldError(
-          'ERR_UNSUPPORTED_LITERAL_TYPE',
-          '"typeTable" must not contain XSD integers, doubles, or booleans.');
+      if(registryEntryId === 1) {
+        // use default table (empty) for registry entry ID `1`
+        typeTable = createTypeTable({typeTable});
+      } else {
+        if(typeof typeTableLoader !== 'function') {
```

Comment:
Looks like maybe the table load and check code could be shared between encode and decode.

Reply:
Yeah, I'll turn this into an issue to do that later.